### PR TITLE
[CouchDB] Renamed CouchDB <database> tag to <dbName> to avoid conflicts

### PR DESCRIPTION
**Pull request by ridz1208**
The <database> tag in config.xml is used for both MySQL database settings and CouchDB. This pull request renames the CouchDB <database> tag to use <dbName> insteal

### CODE REVIEW

File: php/libraries/User.class.inc [hunk 1]
```diff
@@ -401,7 +401,9 @@ class User extends UserPermissions
         $this->update($updateArray);
 
         $config       =& NDB_Config::singleton();
-        $couch_config = $config->getSetting("CouchDB");
+        $couch_config = $config->getSetting('CouchDB');
+
+        //TODO: REMOVE ???
```
Comment (jstirling91) [inline php/libraries/User.class.inc]: I believe there is a pull request to do that, in any case remove this comment

File: php/libraries/CouchDB.class.inc [hunk 2]
```diff
@@ -194,9 +194,9 @@ class CouchDB
     function _constructURL($op, $doc)
     {
         $config = NDB_Config::singleton();
-        $db     = $config->getSetting("CouchDB");
+        $db     = $config->getSetting('CouchDB');
 
-        return "$op /$db[database]/$doc";
+        return "$op /$db[dbName]/$doc";
```
Comment (jstirling91) [inline php/libraries/CouchDB.class.inc]: this needs to be changed in the sample config file

**Comment by jstirling91** [comment]
Adding caveat and add to release notes since all projects will have to update their config files to switch the `database` under the `CouchDB` tag to `dbName`

File: php/libraries/CouchDB.class.inc (hunk 2 above)
Comment (jstirling91) [inline php/libraries/CouchDB.class.inc]: I don't think the file has the basic tags need for couchDB, can you add those too

**Comment by driusan** [comment]
Should this go to 17.1 or 18.0? Discuss.

File: php/libraries/CouchDB.class.inc (hunk 2 above)
Comment (ridz1208) [inline php/libraries/CouchDB.class.inc]: @jstirling91 done and done !

File: docs/config/config.xml [hunk 3]
```diff
@@ -26,6 +26,11 @@
     </database>
 
     <CouchDB>
+        <dbName>loris</dbName>
```
Comment (jstirling91) [inline docs/config/config.xml]: Can you make these similar to the database variables, the install script will soon do something similar to what is done for configuring the database

**Comment by ridz1208** [comment]
@jstirling91 changed field names as requested.

File: docs/config/config.xml [hunk 4]
```diff
@@ -26,6 +26,11 @@
     </database>
 
     <CouchDB>
+        <dbName>%COUCH_DATABASE%</dbName>
+        <hostname>%COUCH_HOSTNAME%</hostname>
+        <port>%COUCH_PORT%</port>
+        <admin>%COUCH_USERNAME%</admin>
+        <adminpass>%COUCH_PASSWORD%</adminpass>
```
Comment (driusan) [inline docs/config/config.xml]: Sorry I just saw this now after you've already made this change.. but why does this need to be in the XML at all?

Changing it can't break LORIS, the config module seems like a reasonable place for it (which would make it easier to provide a patch for existing projects.)

Comment (ridz1208) [inline docs/config/config.xml]: @driusan so you want me to romove this or actually create a script to port this data into mysql ?

Comment (driusan) [inline docs/config/config.xml]: I just checked and there's nothing in the default schema config table, so it's fine.. moving it should be a different PR, it's not high priority.

**Comment by ridz1208** [comment]
@driusan will add it to redmine 17.2
